### PR TITLE
Create function coinList() to get list of coins from CryptoCompare API

### PR DESCRIPTION
**Pull request by owencraston**
I needed to get a list of coins from CryptoCompare so I thought I'd abstract it into this library.
- the coin list data is now available at https://min-api.cryptocompare.com/data/all/coinlist so we can get coinList again.
- more information can be found here:
https://www.cryptocompare.com/api/#-api-data-coinlist-
- write tests for this function
- add documentation for this function
- updated CHANGELOG

cc @RyanZim

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ function fetchJSON (url) {
     })
 }
 
+function coinList() {
```
Comment (RyanZim) [inline index.js]: This function needs to be exported at the bottom of the file.

File: test/cryptocompare.test.js [hunk 2]
```diff
@@ -5,6 +5,17 @@ if (!global.fetch) global.fetch = require('node-fetch')
 const cc = require('../')
 const helpers = require('./helpers')
 
+test('coinList()', t => {
+  t.plan(2)
```
Comment (RyanZim) [inline test/cryptocompare.test.js]: This will make tests fail.

**Comment by owencraston** [comment]
@RyanZim tests seem to still be failing? Is this a problem with the node version?

**Comment by RyanZim** [comment]
@owencraston The linter errors still aren't fixed, please run `npm test` locally and get everything fixed up before pushing here.

File: test/cryptocompare.test.js [hunk 3]
```diff
@@ -5,6 +5,17 @@ if (!global.fetch) global.fetch = require('node-fetch')
 const cc = require('../')
 const helpers = require('./helpers')
 
+test('coinList()', t => {
+  t.plan(4)
+  cc.coinList().then(coins => {
+    t.strictEqual(typeof coins.Data.ETH.Name, 'string', 'coins.Data.ETH.Name` is a string')
+    t.strictEqual(typeof coins.Data.ETH.Name, 'ETH', 'coins.Data.ETH.Name is \'ETH\'')
+    t.strictEqual(typeof coins.Data.ETH.FullName, 'Ethereum (ETH)', 'coins.Data.ETH.FullName is \'Ethereum (ETH)\'')
+    t.strictEqual(typeof coins.Data.ETH.Sponsored, false, 'coins.Data.ETH.Sponsored is \'false\'')
```
Comment (RyanZim) [inline test/cryptocompare.test.js]: You don't want `typeof` here, except for the top assertion.

**Comment by owencraston** [comment]
Based on this doc I think I should remove it: https://docs.npmjs.com/files/package-lock.json
```One key detail about package-lock.json is that it cannot be published, and it will be ignored if found in any place other than the toplevel package. It shares a format with npm-shrinkwrap.json, which is essentially the same file, but allows publication. This is not recommended unless deploying a CLI tool or otherwise using the publication process for producing production packages.```

**Comment by owencraston** [comment]
Why are these tests still failing?

File: index.js [hunk 4]
```diff
@@ -12,6 +12,11 @@ function fetchJSON (url) {
     })
 }
 
+function coinList () {
+  const url = `${baseUrl}coinlist/`
```
Comment (RyanZim) [inline index.js]: Should be:

    const url = `${baseUrl}all/coinlist`

Comment (owencraston) [reply index.js]: Right. I made this change but the tests are still failing.

**Comment by RyanZim** [comment]
@owencraston You need to remove the trailing slash from the URL. CryptoCompare doesn't like it.

**Comment by owencraston** [comment]
Thanks for all the help @RyanZim. Sorry this took so long. I think it should be good to go!

**Comment by jprichardson** [comment]
@RyanZim please release update to npm.

**Comment by jprichardson** [comment]
Thanks @owencraston!

**Comment by owencraston** [comment]
🚀

**Comment by RyanZim** [comment]
Released as 0.3.0 :tada:

**Pull request merged**